### PR TITLE
Adamlevy/scio es5

### PR DESCRIPTION
**Pull request by Alevy01**
Elasticsearch5 support

### CODE REVIEW

**Comment by nevillelyh** [comment]
Are Elasticsearch 2 & 5 compatible with one another? I thought you're gonna great a separate sub project for this?

**Comment by andrewsmartin** [comment]
Also, seems like some changes from `ClosureTest` got muddled in with your changes - do you mind rebasing against master?

**Comment by regadas** [comment]
ES 5 client is not compatible with 2.X cluster. +1 on a sub project for this

File: scio-elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java [hunk 1]
```diff
@@ -204,102 +221,168 @@ public void processElement(ProcessContext c) throws Exception {
...
+    /**
+     * Accepts a Client, a BulkRequestBuilder, and an ActionRequest.
+     * Properly typecasts & prepares an ActionRequest. Adds it to the BulkRequestBuilder.
+     * Return type is void.
+     *
+     * @param client        provides the ability to prepare requests for bulk execution
+     * @param bulkRequest   maintains an ordered list of ActionRequests to later be executed in bulk
+     * @param actionRequest the ActionRequest to be added to the bulkRequest
+     */
+    private static void addBulkRequest(Client client, BulkRequestBuilder bulkRequest,
```
Comment (regadas) [inline scio-elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java]: ES lib already does it https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java#L109

**Comment by Alevy01** [comment]
work in progress, wasn't ready for public PR yet.

**Pull request closed without merging**